### PR TITLE
[Applab Datasets] Modal Pickers for getColumn block

### PR DESCRIPTION
**Pull request by ajpal**
Opens a modal picker for the parameters of the getColumn block.
The options for the table parameter are all of the tables imported in the project.

image

The options for the column parameter are all of the columns in the selected table. 
image

image

If there are no tables in the project:
image

If there are no columns in the table:
image

## Links

- [jira](https://codedotorg.atlassian.net/browse/STAR-1144)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -231,8 +231,10 @@
   "changeUserTypeModal_unexpectedError": "An unexpected error has occurred.  Please wait a moment and try again.",
   "choose": "Choose",
   "chooseActivity": "Choose from the following activities:",
+  "chooseColumn": "Choose a column from {table}",
```
Comment (maddiedierker) [inline apps/i18n/common/en_us.json]: nit: should we add quotes around `{table}` so it reads as `Choose a column from "My Table"`?

File: apps/src/storage/GetColumnParamPicker.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import BaseDialog from '@cdo/apps/templates/BaseDialog.jsx';
+
+export const ParamType = {
+  TABLE: 'TABLE',
+  COLUMN: 'COLUMN'
+};
+
+const styles = {
+  title: {
+    paddingLeft: '1em',
```
Comment (maddiedierker) [inline apps/src/storage/GetColumnParamPicker.jsx]: nit: i think our convention is to use pixels over ems

File: apps/src/storage/GetColumnParamPicker.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    if (this.props.param === ParamType.TABLE) {
+      const reduxState = getStore().getState();
+      options = Object.keys(reduxState.data.tableListMap);
+    } else if (this.props.param === ParamType.COLUMN) {
+      options = this.state.columns;
+    }
+    return (
+      <BaseDialog isOpen useUpdatedStyles handleClose={this.props.onClose}>
+        <h1 style={styles.title}>
+          {this.props.param === ParamType.TABLE
```
Comment (maddiedierker) [inline apps/src/storage/GetColumnParamPicker.jsx]: if we were to add a third ParamType, the ternary on this line and line 75 (error state) would have an unexpected value. i like what you did with `options` above, and i think we could do the same here. then, if we add another ParamType, we can just add to the conditional above and set the appropriate values there

File: apps/src/storage/getColumnDropdown.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    var options = [];
+    options.push({
+      text: msg.choosePrefix(),
+      display:
+        '<span class="chooseAssetDropdownOption">' +
+        msg.choosePrefix() +
+        '</a>',
+      click: callback => openModal(ParamType.TABLE, callback)
+    });
+    return options;
```
Comment (maddiedierker) [inline apps/src/storage/getColumnDropdown.js]: i might be missing a convention here, but could this simplify to:

```
return [{
  text: msg.choosePrefix(),
  display:
    '<span class="chooseAssetDropdownOption">' +
      msg.choosePrefix() +
      '</a>',
  click: callback => openModal(ParamType.TABLE, callback)
}];
```

Comment (maddiedierker) [reply apps/src/storage/getColumnDropdown.js]: (same for `getColumns()` below)

Comment (ajpal) [reply apps/src/storage/getColumnDropdown.js]: I was using getAssetDropdown as a model, but yeah that's much simpler for this case.

**Review by maddiedierker — APPROVED** [review]

**Pull request merged**